offchain-modules/packages/x/src/handlers/eth.ts

```diff
@@ -77,9 +78,18 @@ export class EthHandler {
   }
 
   async watchNewBlock() {
-    await this.ethChain.watchNewBlock(this.lastHandledBlockHeight, async (newBlock: ethers.providers.Block) => {
-      await this.onBlock(newBlock);
-    });
+    for (let i = 0; i <= MAX_RETRY_TIMES; i++) {
+      try {
+        await this.ethChain.watchNewBlock(this.lastHandledBlockHeight, async (newBlock: ethers.providers.Block) => {
```

Comment:
try-catch should add into ethChain.watchNewBlock, and retry should add into this.onBlock, I think.